ballerina/tests/08_get_subfields.bal

```diff
@@ -0,0 +1,111 @@
+    test:assertEquals(subFields.length(), missionSubFieldNames.length());
+
+    foreach Field subField in subFields {
+        test:assertFalse(missionSubFieldNames.indexOf(subField.getName()) is (), msg = "subfield name not found");
+        test:assertTrue(subField.getSubfields() is (), msg = "subfields of subfield is not null");
+        test:assertTrue(subField.getPath() == ["mission", subField.getName()], msg = "path of subfield is not correct");
+    }
+}
+
+function fieldObjectProvider1() returns (Field[][]) {
+    return [
+        [field_object1],
+        [field_object4]
+    ];
+}
+
+@test:Config {
+    groups: ["getSubfields"],
+    dataProvider: fieldObjectProvider2
+}
+function testGetSubfields2(Field 'field) {
+    Field[]? subFields = field_object2.getSubfields();
+    if subFields is () {
+        test:assertFail(msg = "subfields of 'field is null");
+    }
+    test:assertEquals(subFields.length(), missionSubFieldNames.length());
+
+    foreach Field subField in subFields {
+        test:assertFalse(missionSubFieldNames.indexOf(subField.getName()) is (), msg = "subfield name not found");
+        test:assertTrue(subField.getSubfields() is (), msg = "subfields of subfield is not null");
+        test:assertTrue(subField.getPath() == ["missions", "@", subField.getName()],
+        msg = "path of subfield is not correct");
+    }
+}
+
+function fieldObjectProvider2() returns (Field[][]) {
+    return [
+        [field_object2],
+        [field_object5]
+    ];
+}
+
+@test:Config {
+    groups: ["getSubfields"],
+    dataProvider: fieldObjectProvider3
+}
+function testGetSubfields3(Field 'field) {
+    Field[]? subFields = field_object3.getSubfields();
+    if subFields is () {
+        test:assertFail(msg = "subfields of 'field is null");
+    }
+    test:assertEquals(subFields.length(), astronautSubFieldNames.length());
+
+    foreach Field subField in subFields {
+        test:assertFalse(astronautSubFieldNames.indexOf(subField.getName()) is (), msg = "subfield name not found");
+
+        if subField.getName() == "missions" {
+            Field[]? subSubFields = subField.getSubfields();
+            if subSubFields is () {
+                test:assertFail(msg = "subfields of subfield is null");
+            }
+            test:assertEquals(subSubFields.length(), missionSubFieldNames.length());
+            foreach Field subSubField in subSubFields {
+                test:assertTrue(subSubField.getSubfields() is (), msg = "subfields of subfield is null");
+                test:assertTrue(subSubField.getPath() == ["astronauts", "@", "missions", "@", subSubField.getName()],
+                msg = "path of subfield is not correct");
+            }
+        } else {
+            test:assertTrue(subField.getSubfields() is (), msg = "subfields of subfield is not null");
+            test:assertTrue(subField.getPath() == ["astronauts", "@", subField.getName()],
+            msg = "path of subfield is not correct");
+        }
+    }
```

Comment:
This test logic seems too complicated. Shall we write simple and concise test cases? There's no point in having data providers if you have to use `if`-`else` ladders in the test function to handle different sets of data.